benchmark/test_special_perf.py

```diff
@@ -244,3 +245,23 @@ def upsample_nearest2d_input_fn(shape, dtype, device):
         dtypes=FLOAT_DTYPES,
     )
     bench.run()
+
+
+@pytest.mark.diag
+def test_perf_diag():
+    def diag_input_fn(shape, dtype, device):
+        if shape[0] < 10000:
```

Comment:
why require shape[0] less than 10000? test shapes in GenericBenchmardExcluse3D all satisfy.